pages/dao/[symbol]/params/GovernanceConfigModal.tsx

```diff
@@ -18,93 +18,173 @@ import useQueryContext from '@hooks/useQueryContext'
 import { useRouter } from 'next/router'
 import { notify } from '@utils/notifications'
 import useRealm from '@hooks/useRealm'
-import { useState } from 'react'
+import { useEffect, useState } from 'react'
 import Button from '@components/Button'
 import BaseGovernanceForm, {
-  BaseGovernanceFormFields,
+  BaseGovernanceFormFieldsV2,
 } from '@components/AssetsList/BaseGovernanceForm'
-import { getGovernanceConfig } from '@utils/GovernanceTools'
+import { getGovernanceConfigFromV2Form } from '@utils/GovernanceTools'
 import {
   getDaysFromTimestamp,
   getMintDecimalAmountFromNatural,
 } from '@tools/sdk/units'
 import { abbreviateAddress } from '@utils/formatting'
 import * as yup from 'yup'
 import { DISABLED_VOTER_WEIGHT } from '@tools/constants'
+import useProgramVersion from '@hooks/useProgramVersion'
+import {
+  BaseGovernanceFormFieldsV3,
+  transformerGovernanceConfig_2_BaseGovernanceFormFieldsV3,
+  transform,
+  transformerBaseGovernanceFormFieldsV3_2_GovernanceConfig,
+} from '@components/AssetsList/BaseGovernanceForm-data'
 
-interface GovernanceConfigForm extends BaseGovernanceFormFields {
+type ProposalInfo = {
   title: string
   description: string
 }
 
+type GovernanceConfigFormV2 = BaseGovernanceFormFieldsV2 & ProposalInfo
+type GovernanceConfigFormV3 = BaseGovernanceFormFieldsV3 & ProposalInfo
+
+type ModalProps = {
+  closeProposalModal: () => void
+  isProposalModalOpen: boolean
+}
+
+type Props = {
+  governance: ProgramAccount<Governance>
+} & ModalProps
+
+const DEFAULT_PROPOSAL_TITLE = 'Change governance config'
+
 const GovernanceConfigModal = ({
   closeProposalModal,
   isProposalModalOpen,
   governance,
-}: {
-  closeProposalModal: () => void
-  isProposalModalOpen: boolean
-  governance: ProgramAccount<Governance>
-}) => {
+}: Props) => {
   const router = useRouter()
-  const { realm, canChooseWhoVote, symbol, mint, realmInfo } = useRealm()
+  const {
+    realm,
+    canChooseWhoVote,
+    symbol,
+    mint,
+    realmInfo,
+    councilMint,
+  } = useRealm()
   const config = governance?.account.config
   const { fmtUrlWithCluster } = useQueryContext()
   const wallet = useWalletStore((s) => s.current)
   const { handleCreateProposal } = useCreateProposal()
-  const defaultCfgTitle = 'Change governance config'
   const [formErrors, setFormErrors] = useState({})
   const [creatingProposal, setCreatingProposal] = useState(false)
   const [voteByCouncil, setVoteByCouncil] = useState(false)
-  const [form, setForm] = useState<GovernanceConfigForm>({
```

Comment:
https://medium.com/@digruby/do-not-use-props-as-default-value-of-react-usestate-directly-818ee192f454